server/datastore/mysql/hosts.go

```diff
@@ -851,7 +851,8 @@ func (d *Datastore) SaveHostUsers(host *fleet.Host) error {
 		incomingUsers[u.Uid] = true
 
 		if _, err := d.writer.Exec(
-			`INSERT IGNORE INTO host_users (host_id, uid, username, user_type, groupname) VALUES (?, ?, ?, ?, ?)`,
+			`INSERT IGNORE INTO host_users (host_id, uid, username, user_type, groupname) VALUES (?, ?, ?, ?, ?)
+				ON DUPLICATE KEY UPDATE removed_at=NULL`,
```

Comment:
:D Nice! Is `IGNORE` still needed in that case?

Reply:
Only to silent errors such as a potential duplicated user being reported. We might want to see that error though?

Reply:
The `ON DUPLICATE KEY UPDATE` section would eat up that error (and turn it into an update).  I believe `IGNORE` also skips other errors such as data conversion or NULL in NOT NULL, which we probably don't want to ignore.

Reply:
Agreed! I'll address this when I resolve the conflict after merging the other PR.

Reply:
Good call, @mna.